File: 8kyu/(8 kyu) MakeUpperCase/(8 kyu) MakeUpperCase.go
package kata

// 1 Simple solution
func MakeUpperCase(str string) string {
	res := ""
	for _, char := range str {
		if char >= 'a' && char <= 'z' {
			res += string(char - 32)
		} else {
			res += string(char)
		}
	}
	return res
}

// 2 Optimized solution
import "regexp"
func MakeUpperCase(str string) string {
	var re = regexp.MustCompile("[a-z]")
	return re.ReplaceAllStringFunc(str, func(s string) string {
		return string(s[0] - 32)
	})
}

// 3 Clever solution
import "strings"

func MakeUpperCase(str string) string {
	return strings.ToUpper(str)
}

File: 8kyu/(8 kyu) MakeUpperCase/(8 kyu) MakeUpperCase.cs
// 1 Simple solution
public class Kata
{
    public static string MakeUpperCase(string str)
    {
        string result = "";
        for (int i = 0; i < str.Length; i++)
        {
            char c = str[i];
            if (c >= 'a' && c <= 'z')
            {
                result += (char)(c - ('a' - 'A'));
            }
            else
            {
                result += c;
            }
        }
        return result;
    }
}

// 2 Optimized solution
using System.Linq;

public class Kata
{
    public static string MakeUpperCase(string str) =>
        str.
            ToCharArray()
                .Select(c => c >= 'a' && c <= 'z' ? (char)(c - ('a' - 'A')) : c)
                .Aggregate("", (acc, c) => acc + c);
}

// 3 Clever solution
public class Kata
{
    public static string MakeUpperCase(string str) => str.ToUpper();
}
